charts/core/src/Grid/config/useGridOptions.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react';
+
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { color, InteractionState, Variant } from '@leafygreen-ui/tokens';
+
+import { ChartOptions } from '../../Chart';
+import { GridProps } from '../Grid.types';
+
+export const useGridOptions = ({ horizontal, vertical }: GridProps) => {
+  const [options, setOptions] = useState<Partial<ChartOptions>>({});
```

Comment:
If we moved the `updateChartOptions` call into this hook (or the inverse) I don't think we'd need this state object.